src/coreclr/jit/objectalloc.cpp

```diff
@@ -712,12 +804,30 @@ void ObjectAllocator::UpdateAncestorTypes(GenTree* tree, ArrayStack<GenTree*>* p
                 if (parent->TypeGet() == TYP_REF)
                 {
                     parent->ChangeType(newType);
+
+                    if (parent->OperIs(GT_COLON))
+                    {
+                        GenTree* const lhs = parent->AsOp()->gtGetOp1();
+                        GenTree* const rhs = parent->AsOp()->gtGetOp2();
+
+                        if (lhs->TypeIs(TYP_REF))
+                        {
+                            lhs->ChangeType(newType);
+                        }
+
+                        if (rhs->TypeIs(TYP_REF))
+                        {
+                            rhs->ChangeType(newType);
+                        }
+                    }
```

Comment:
Seems a bit iffy to me to change the type of arbitrary operand nodes.. is that ok?

Reply:
Yeah this should be handled differently. Could be the other sibling really is a ref and we need to retype upwards as byref and not nint.